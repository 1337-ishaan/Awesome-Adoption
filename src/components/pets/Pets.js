import React from "react";
import { Row, Col, Image, Button } from "react-bootstrap";

import "./pets.css";
import Bird from "./bird.jpg";
import Dog from "./doggo.jpg";
import Cat from "./cat.jpg";
import Horse from "./horse.jpg";
import Rabbit from "./rabbit.jpg";
import { Link, useRouteMatch } from "react-router-dom";
export default function Pets() {
  //   const [pets, setPets] = useState("");

  //   useEffect(() => {
  //     const config = {
  //       headers: { Authorization: `Bearer ${token}` },
  //     };
  //     axios
  //       .get("https://api.petfinder.com/v2/types", config)
  //       .then((response) => {
  //         setPets(response.data);
  //       })
  //       .catch((error) => {
  //         console.log(error);
  //       });
  //   }, [token]);

  let { url } = useRouteMatch();
  return (
    <div className="pet__container">
      <h1>Adopt Your Buddy</h1>
      <Row>
        <AnimalType img={Dog} type="dog" link={`${url}/dog`} />
        <AnimalType img={Cat} type="cat" link={`${url}/cat`} />
        <AnimalType img={Bird} type="bird" link={`${url}/bird`} />
        <AnimalType img={Horse} type="horse" link={`${url}/horse`} />
        <AnimalType img={Rabbit} type="rabbit" link={`${url}/rabbit`} />
      </Row>
    </div>
  );
}

const AnimalType = ({ type, img, link }) => (
  <Col xs={12} md={6} className="pet__column">
    <Image src={img} alt={type} roundedCircle />
    <Button as={Link} to={link} className="pet__button" color="primary">
      Click Here
    </Button>
  </Col>
);
