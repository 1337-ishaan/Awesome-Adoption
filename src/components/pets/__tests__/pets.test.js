import { cleanup, render, screen } from "@testing-library/react";
import Pets, { AnimalType } from "../Pets";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
afterEach(() => {
  cleanup();
});

test("matches animal link snapshot", () => {
  const animalLink = {
    img: "https://via.placeholder.com/150",
    type: "playtpus",
    link: "pets/playtpus",
  };
  const tree = renderer
    .create(
      <BrowserRouter>
        <AnimalType
          type={animalLink.type}
          img={animalLink.img}
          link={animalLink.link}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
