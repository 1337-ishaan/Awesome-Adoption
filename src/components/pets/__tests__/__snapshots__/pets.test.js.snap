// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches animal link snapshot 1`] = `
<div
  className="pet__column d-flex flex-column align-items-center justify-content-end my-4 col-md-6 col-12"
>
  <img
    alt="playtpus"
    className="rounded-circle"
    src="https://via.placeholder.com/150"
  />
  <a
    className="pet__button btn btn-primary"
    color="primary"
    disabled={false}
    href="/pets/playtpus"
    onClick={[Function]}
  >
    Click Here
  </a>
</div>
`;
